From e58dc00d30f9e9def4fa9a9e21b1b46090d89e41 Mon Sep 17 00:00:00 2001
From: Johannes Lorenz <j.git@lorenz-ho.me>
Date: Fri, 18 Dec 2020 18:48:13 +0100
Subject: [PATCH 1684/1936] FL UI: Add "fall through" markers

The code here is very old and the fall throughs have been seen by many
coders. This fact, and the code itself, imply that the fall throughs are
very likely intended.
---
 src/UI/EnvelopeFreeEdit.cpp | 1 +
 src/UI/Fl_Osc_Slider.cpp    | 5 +++--
 src/UI/Fl_Osc_TSlider.cpp   | 1 +
 src/UI/WidgetPDial.cpp      | 1 +
 4 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/UI/EnvelopeFreeEdit.cpp b/src/UI/EnvelopeFreeEdit.cpp
index e3e4737d..2cd1a981 100644
--- a/src/UI/EnvelopeFreeEdit.cpp
+++ b/src/UI/EnvelopeFreeEdit.cpp
@@ -283,6 +283,7 @@ int EnvelopeFreeEdit::handle(int event)
               if (pair!=NULL) pair->redraw();
               return 1;
           }
+          // fall through
       case FL_DRAG:
           if (currentpoint>=0){
               old_mod_state = mod_state;
diff --git a/src/UI/Fl_Osc_Slider.cpp b/src/UI/Fl_Osc_Slider.cpp
index b606d846..076269a5 100644
--- a/src/UI/Fl_Osc_Slider.cpp
+++ b/src/UI/Fl_Osc_Slider.cpp
@@ -70,8 +70,8 @@ void Fl_Osc_Slider::cb(void)
     else if(osc_type == 'i')
         oscWrite(ext, "i", (int)(val-min_));
     else {
-	fprintf(stderr, "invalid `c' from slider %s%s, using `i'\n", loc.c_str(), ext.c_str());
-	oscWrite(ext, "i", (int)(val-min_));
+        fprintf(stderr, "invalid `c' from slider %s%s, using `i'\n", loc.c_str(), ext.c_str());
+        oscWrite(ext, "i", (int)(val-min_));
     }
     //OSC_value(val);
 
@@ -132,6 +132,7 @@ int Fl_Osc_Slider::handle(int ev, int X, int Y, int W, int H)
                         break;
                     case FL_CTRL:
                         divisor = 128;
+                        // fall through
                     default:
                         step_ = absrange / divisor;
                         if (step_ < 1)
diff --git a/src/UI/Fl_Osc_TSlider.cpp b/src/UI/Fl_Osc_TSlider.cpp
index 112184ca..ed33ab1c 100644
--- a/src/UI/Fl_Osc_TSlider.cpp
+++ b/src/UI/Fl_Osc_TSlider.cpp
@@ -54,6 +54,7 @@ int Fl_Osc_TSlider::handle(int event)
                 return(1);
             tipwin->position(Fl::event_x_root()-Fl::event_x()+x(),
                              Fl::event_y_root()-Fl::event_y()+h()+y()+5);
+            // fall through
         case FL_DRAG:
             tipwin->showValue(transform(value()));
             break;
diff --git a/src/UI/WidgetPDial.cpp b/src/UI/WidgetPDial.cpp
index db597ad8..e6830689 100644
--- a/src/UI/WidgetPDial.cpp
+++ b/src/UI/WidgetPDial.cpp
@@ -84,6 +84,7 @@ int WidgetPDial::handle(int event)
             }
             oldvalue = value();
             old_y = Fl::event_y();
+            // fall through
         case FL_DRAG:
             getPos();
             old_mod_state = mod_state;
-- 
2.47.0

